### PR TITLE
Add async version of specialized form parser

### PR DESCRIPTION
**Pull request by asrivas**

### CODE REVIEW

File: specialized/async_specialized_form_parser.ipynb [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+    "id": "_v0XtSwn7fmN",
+    "outputId": "373b9379-f6ac-451a-e428-ad9219fb31d0"
+   },
+   "outputs": [],
+   "source": [
+    "# Install necessary Python libraries\n",
+    "!pip install google-cloud-documentai\n",
+    "!pip install 'google-cloud-storage==1.31.0'\n",
+    "\n",
+    "!apt-get update"
```
Comment (camushoilingma) [inline specialized/async_specialized_form_parser.ipynb:29]: Do we really need this like "!apt-get update"

File: specialized/async_specialized_form_parser.ipynb [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+    "    input_config = documentai.types.document_processor_service.BatchProcessRequest.BatchInputConfig(\n",
+    "        gcs_source=GCS_INPUT_URI, mime_type=\"application/pdf\"\n",
+    "    )\n",
+    "\n",
+    "    # Where to write results\n",
+    "    output_config = documentai.types.document_processor_service.BatchProcessRequest.BatchOutputConfig(\n",
+    "        gcs_destination=destination_uri\n",
+    "    )\n",
+    "\n",
+    "    # Location can be 'us' or 'eu'\n",
```
Comment (camushoilingma) [inline specialized/async_specialized_form_parser.ipynb:106]: Comment redundant. In the other notebooks. FWIW comment was:
    # The full resource name of the processor, e.g.:
    # projects/project-id/locations/location/processor/processor-id
    # You must create new processors in the Cloud Console first

File: specialized/async_specialized_form_parser.ipynb [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+    "    request = documentai.types.document_processor_service.BatchProcessRequest(\n",
+    "        name=name,\n",
+    "        input_configs=[input_config],\n",
+    "        output_config=output_config,\n",
+    "    )\n",
+    "\n",
+    "    operation = client.batch_process_documents(request)\n",
+    "\n",
+    "    # Wait for the operation to finish\n",
+    "    operation.result(timeout=timeout)\n",
```
Comment (camushoilingma) [inline specialized/async_specialized_form_parser.ipynb:117]: "timeout" is never defined. Compilation error.

**Review by camushoilingma — CHANGES_REQUESTED** [review]
I get compilation errors because the timeout variable isn't defined. 

Can we use directory with multiple invoices as an example?

File: specialized/async_specialized_form_parser.ipynb (hunk 3 above)
Comment (camushoilingma) [reply specialized/async_specialized_form_parser.ipynb:117]: Maybe set the TIMEOUT=300 above.

Here add a comment like # Operation timeout in seconds. Default is 120s.

File: specialized/async_specialized_form_parser.ipynb (hunk 1 above)
Comment (camushoilingma) [reply specialized/async_specialized_form_parser.ipynb:29]: Instead we should either

1) Let's add a requirements.txt in the top directory with the couple lines below and add instructions to download the libraries in the README?

google-cloud-documentai==0.3.0
google-cloud-storage==1.36.2

2) Add a text above saying "Install the libraries and restart the kernel." Put this all under a "Setup" section. We can also add some validations like:

Validate storage version is >=1.36.2
`!python3 -c "from google.cloud import storage; print('Storage version: {}'.format(storage.__version__))"`

**Review by camushoilingma — CHANGES_REQUESTED** [review]

File: specialized/async_specialized_form_parser.ipynb (hunk 1 above)
Comment (asrivas) [reply specialized/async_specialized_form_parser.ipynb:29]: First option sounds great, will add requirements.txt in the PR

File: specialized/async_specialized_form_parser.ipynb (hunk 2 above)
Comment (asrivas) [reply specialized/async_specialized_form_parser.ipynb]: Great catch -- thank you, adding back the original comment

**Review by asrivas — COMMENTED** [review]
Great feedback, I've made some of the changes but not all. Let me grab sometime with you on Monday to make sure I get the set up instructions and requirements.txt added correctly so I can finish up the PR.

File: README.md [hunk 4]
```diff
@@ -18,6 +18,8 @@ You must have your own GCP project with billing enabled and have working knowled
 1. Set up your GCP project for Document AI following the [Setup Guide][set_up].
 1. Enable the 'Document AI API' in your project in the Document AI [Platform][platform].
 1. [Create][create_notebook] or use an existing instance of AI Platform Notebook with Python 3 using the default configurations.
+1. In the notebook, go to **Git** > **Clone a Repository** and paste the repository URL.
+1. Install the libraries from the `requirements.txt`.
```
Comment (camushoilingma) [inline README.md:22]: python -m pip install -r requirements.txt

File: specialized/async_specialized_form_parser.ipynb (hunk 1 above)
Comment (camushoilingma) [reply specialized/async_specialized_form_parser.ipynb:29]: works for me! but maybe say here to make sure that they run python -m pip install -r  requirements.txt from the root directory.

Comment (camushoilingma) [reply specialized/async_specialized_form_parser.ipynb:29]: or add this as a cell !python -m pip install -r ../requirements.txt

Comment (asrivas) [reply specialized/async_specialized_form_parser.ipynb]: Did the API change between 1.31 & 1.36.2? I'm getting:
TypeError: __init__() got an unexpected keyword argument 'client_options'
on the storage client with 1.36

File: specialized/async_specialized_form_parser.ipynb (hunk 2 above)
Comment (asrivas) [reply specialized/async_specialized_form_parser.ipynb]: Created gs://cloud-samples-data/documentai/async_invoices/ with 3, will add the GCS code to iterate through the folder and create the input configs tomorrow

File: requirements.txt [hunk 5]
```diff
@@ -0,0 +1,3 @@
+google-cloud-core==1.6.0
+google-cloud-documentai==0.3.0
+google-cloud-storage==1.36.2
```
Comment (asrivas) [inline requirements.txt:3]: I tested out the bounding boxes in the generic form parser a couple times, I *believe* we don't need any additional packages here

Comment (asrivas) [reply requirements.txt:3]: also will add the bounding boxes back to the specialize parser in a follow up PR

**Comment by camushoilingma** [comment]
LGTM, just remove "en" from PDF_PATH = "../resources/procurement/invoices/en/invoice.pdf" in sync notebook if possible. Thanks for the great additions!

**Review by camushoilingma — APPROVED** [review]

**Comment by asrivas** [comment]
> LGTM, just remove "en" from PDF_PATH = "../resources/procurement/invoices/en/invoice.pdf" in sync notebook if possible. Thanks for the great additions!

I think you got that change into master with #10, it's not showing up in my branch anymore after syncing to head. If after I push it's there I'll submit a new PR.

**Pull request merged**